### PR TITLE
feat: Add support for tel: and mailto: URIs in Advanced Page Settings redirect field

### PR DESCRIPTION
**Pull request by marksweb**
## Description

This re-implements the linked stale PR as this functionality seems useful.

## Related resources

* #6568

## Checklist

* [x] I have opened this pull request against ``develop``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: cms/forms/validators.py [hunk 1]
```diff
@@ -22,6 +24,20 @@ def validate_url(value):
...
 
+def validate_url_extra(value):
+    try:
+        # Try normal validator first
+        validate_url(value)
+    except ValidationError:
+        # Fallback to absolute urls
+        if re.match(r'^tel:[0-9\+\#\*\-\.\(\)]+$', value):
+            pass
+        elif re.match(r'^mailto:\S+$', value):
```
Comment (fsbraun) [inline cms/forms/validators.py:35]: I'm totally not into regex at all. Does this check if after `mailto:` there follows a valid email address?

File: cms/forms/validators.py [hunk 2]
```diff
@@ -22,6 +24,20 @@ def validate_url(value):
         URLValidator()(value)
 
 
+def validate_url_extra(value):
+    try:
+        # Try normal validator first
+        validate_url(value)
+    except ValidationError:
+        # Fallback to absolute urls
+        if re.match(r'^tel:[0-9\+\#\*\-\.\(\)]+$', value):
```
Comment (fsbraun) [inline cms/forms/validators.py:33]: Validating phone numbers internationally is probably a total pain. Unless there is a standard way of doing this, I'd leave it at this.

**Review by fsbraun — COMMENTED** [review]
Hi @marksweb ! I was just wondering of there is a Django or python way to validate the email address following a `mailto:`

**Comment by marksweb** [comment]
@fsbraun the isn't my work. There's a linked PR that's stale & I was making the changes exactly as they were.

**Review by fsbraun — APPROVED** [review]

**Pull request merged**